src/ohio/baseio.py

```diff
@@ -85,3 +86,41 @@ def readline(self):
                 break
 
         return result
+
+
+class StreamTextIOBase(StreamIOBase, io.TextIOBase):
+    """Readable file-like abstract base class for text.
+
+    Concrete classes may implemented method `__next_chunk__` to return
+    chunks (or all) of the text to be read.
+
+    """
+
+    def __init__(self):
+        self._remainder = ''
+
+    @staticmethod
+    def _get_empty_value():
+        return ''
+
+    @staticmethod
+    def _get_newline():
+        return '\n'
+
+
+class StreamBufferedIOBase(StreamIOBase, io.BufferedIOBase):
```

Comment:
Yep, I reviewed the docs too, just didn't think it mattered. But thanks, I see what you mean now.

I had been meaning to play around with stuff like that, namely giving it a base of say `IOBase` (rather than `TextIOBase`) – #3. Really, it's just a pass-through, so it doesn't seem like it should care either way; (and, that's why I like the lazy type detection version, discussed below).